src/features/embalm/stepContent/components/FilterInput.tsx

```diff
@@ -32,93 +45,85 @@ export function FilterInput({ filterName, placeholder = '', ...rest }: FilterPro
+  icon,
+  ...rest
+}: FilterComponentProps) {
+  const dispatch = useDispatch();
 
-    return (
-      <Flex align="center">
-        <InputGroup>
-          <NumberInput
-            w="150px"
-            onChange={handleChangeDiggingFees}
-            {...rest}
-          >
-            <NumberInputField
-              pl={12}
-              pr={1}
-              placeholder={placeholder}
-              borderColor="violet.700"
-            />
+  return (
+    <Flex align="center">
+      <InputGroup>
+        <NumberInput
+          w={filterWidth}
+          onChange={(valueAsString: string, valueAsNumber: number) =>
+            dispatch(validateAndSetInput(valueAsString, valueAsNumber, filterTypeAction))
+          }
+          {...rest}
+        >
+          <NumberInputField
+            pl={icon ? '12' : '3'}
+            pr={1}
+            placeholder={placeholder}
+            borderColor="violet.700"
+          />
+          {icon && (
             <InputLeftElement>
               <SarcoTokenIcon boxSize="16px" />
             </InputLeftElement>
-          </NumberInput>
-        </InputGroup>
-      </Flex>
-    );
-  } else if (filterName === 'Unwraps') {
-    function setUnwraps(unwraps: string) {
-      return dispatch(setUnwrapsFilter(unwraps));
-    }
+          )}
+        </NumberInput>
+      </InputGroup>
+    </Flex>
+  );
+}
+
+export function FilterInput({ filterName, placeholder = '', ...rest }: FilterProps) {
+  switch (filterName) {
+    case SortFilterType.DIGGING_FEES:
+      return (
+        <FilterComponent
+          filterWidth={'150px'}
+          placeholder={placeholder}
+          filterTypeAction={setDiggingFeesFilter}
+          icon={true}
+          {...rest}
+        />
+      );
 
-    function handleChangeUnwraps(valueAsString: string, valueAsNumber: number) {
-      checkParams(valueAsString, valueAsNumber);
+    case SortFilterType.UNWRAPS:
+      return (
+        <FilterComponent
+          filterWidth={'100px'}
+          placeholder={placeholder}
+          filterTypeAction={setUnwrapsFilter}
+          {...rest}
+        />
+      );
 
-      setUnwraps(valueAsString);
-    }
+    case SortFilterType.FAILS:
+      return (
+        <FilterComponent
+          filterWidth={'82px'}
+          placeholder={placeholder}
+          filterTypeAction={setFailsFilter}
+          {...rest}
+        />
+      );
 
-    return (
-      <Flex align="center">
-        <InputGroup>
-          <NumberInput
-            w="100px"
-            onChange={handleChangeUnwraps}
-            {...rest}
-          >
-            <NumberInputField
-              pl={3}
-              pr={1}
-              placeholder={placeholder}
-              borderColor="violet.700"
-            />
-          </NumberInput>
-        </InputGroup>
-      </Flex>
-    );
-  } else {
-    function setFails(fails: string) {
-      return dispatch(setFailsFilter(fails));
-    }
+    case SortFilterType.ADDRESS_SEARCH:
```

Comment:
This is dealt separately in the useArchaeologistList hook to accomodate the use of letters as input

Reply:
Type is kept to allow toggle of arrow icon on the right when pressing sort